web/beacon-app/src/features/onboarding/components/steps/preference/form.tsx

```diff
@@ -0,0 +1,60 @@
+import { t } from '@lingui/macro';
+import { Form, FormikHelpers, FormikProvider } from 'formik';
+import React from 'react';
+
+import Select from '@/components/ui/Select';
+
+import { usePreferenceForm } from '../../../usePreferenceForm';
+import { getWorksOptions } from '../../../utils';
+import StepButtons from '../../StepButtons';
+import DeveloperSegment from './DeveloperSegment';
+import { ProfessionSegment } from './profession';
+type PreferenceFormProps = {
+  onSubmit: (values: any, helpers: FormikHelpers<any>) => void;
+  isDisabled?: boolean;
+  isSubmitting?: boolean;
+  initialValues?: any;
+};
+const UserPreferenceForm = ({ onSubmit, isSubmitting, initialValues }: PreferenceFormProps) => {
+  const formik = usePreferenceForm(onSubmit, initialValues);
+  const { values, setFieldValue } = formik;
+  const ROLE_OPTIONS = getWorksOptions();
```

Comment:
Will setting the max # of options a user can select to 3 be handled in a future story?

Reply:
I forgot to disable the option when reaching the maximum value. Thanks for bringing this to my attention. I've just added it.